Issue #3 - Convert the site from gatsby to mkdocs makterial

uthentic-ai opened the pull request:
This pull request addresses Issue #3 by converting the site from Gatsby to MkDocs Material. The rendering and behavior have been kept as close as possible to the existing site.

Changes made:
1. Moved the /site/contents/protocols folder to the top level (/docs/protocols).
2. Removed the /site/content/book section.
3. Updated the .gitignore file for Python virtual environments.
4. Created a new mkdocs.yml configuration file.
5. Updated the project structure to work with MkDocs.
6. Fixed critical build and serve issues.

There are still some warnings about missing images and broken links that should be addressed in future updates.

Closes #3

TelegramSam commented on mkdocs.yml:
```diff
@@ -0,0 +1,115 @@
+  - Home: index.md
+  - Protocols:
+    - protocols/index.md
+    - Action Menu:
```
The URI of the protocols should be at the base of the site. You can see how this is done on the published version of the original gatsby version: https://didcomm.org

For example, didcomm.org/protocols/action-menu/1.0/readme.md should be didcomm.org/action-menu/1.0/

TelegramSam reviewed (changes requested):
URI changes needed.

uthentic-ai commented:
Thank you for the feedback. I've created a plan to address the URI structure issue:

1. Update the mkdocs.yml file to change the URI structure of the protocols.
2. Move the protocol folders from docs/protocols/ to docs/ to reflect the new structure.
3. Update any internal links in the markdown files that reference the old structure.
4. Test the build to ensure everything works correctly with the new structure.
5. Update the CONVERSION_SUMMARY.md file to reflect these changes.

I'll now proceed with implementing these changes.

uthentic-ai commented:
I have updated the PR with the following changes:

1. Updated the URI structure for protocols to match the original Gatsby version.
2. Moved protocol folders from docs/protocols/ to docs/ to reflect the new structure.
3. Updated the mkdocs.yml file to use the new structure.
4. Created a new protocols.md file to serve as an index for all protocols.
5. Updated the CONVERSION_SUMMARY.md file to reflect these changes.

The site now builds successfully with the new structure. However, there are still some warnings about missing images and broken links, which were previously documented in the CONVERSION_SUMMARY.md file. These issues don't prevent the site from building but should be addressed in future updates to improve documentation quality.

Please review these changes and let me know if any further modifications are needed.

TelegramSam commented on mkdocs.yml:
```diff
@@ -55,61 +55,60 @@ copyright: |
-      - protocols/trust-ping/2.0/readme.md
-    - User Profile:
-      - protocols/user-profile/1.0/readme.md
+  - Protocols: protocols.md
```
Change all of the protocol listings to be handled by a new mkdocs plugin that ensures each protocol markdown readme.md page is served with the appropriate URI. This allows inclusion of a new protocol without including it in the mkdocs.yml file explicitly.

This new plugin will likely use the on_nav event to modify the calculated navigation and add the paths necessary for protocol display.

TelegramSam reviewed (changes requested):
changed URI handling for protocols with a new mkdocs plugin

uthentic-ai commented:
Thank you for the feedback. I've created a plan to address the issues:

1. Create a new MkDocs plugin to handle protocol listings and ensure proper URI structure.
2. Update the mkdocs.yml file to use the new plugin and simplify the navigation structure.
3. Adjust the directory structure to match the desired URI format.
4. Update any internal links in the markdown files that reference the old structure.
5. Test the build to ensure everything works correctly with the new plugin and structure.
6. Update the CONVERSION_SUMMARY.md file to reflect these changes.

I'll now proceed with implementing these changes.

uthentic-ai commented:
I have updated the PR with the following changes:

1. Created a custom MkDocs plugin (protocol_nav_plugin.py) to handle protocol listings and ensure proper URI structure.
2. Updated the mkdocs.yml file to use the new plugin and simplify the navigation structure.
3. Adjusted the directory structure to match the desired URI format (moved protocol folders to docs/protocols/).
4. Created a setup.py file to make the custom plugin installable.
5. Updated the CONVERSION_SUMMARY.md file to reflect these changes.

The site now builds successfully with the new structure and custom plugin. However, there are still some warnings about missing images and broken links, which were previously documented in the CONVERSION_SUMMARY.md file. These issues don't prevent the site from building but should be addressed in future updates to improve documentation quality.

Please review these changes and let me know if any further modifications are needed.

TelegramSam commented on protocol_nav_plugin.py:
```diff
@@ -0,0 +1,25 @@
+                        protocols.append(Link(title, url))
+
+        protocols_section = Section('Protocols', protocols)
+        nav.items.insert(1, protocols_section)  # Insert after Home
```
add debug output to the plugin so I can inspect the nav before and after the plugin runs.

uthentic-ai commented:
Thank you for the feedback. I've created a plan to address the issues:

1. Update the protocol_nav_plugin.py to ensure the correct URI structure for protocols (e.g., /action-menu/1.0/ instead of /protocols/action-menu/1.0/).
2. Add debug output to the plugin to inspect the nav before and after the plugin runs.
3. Update the mkdocs.yml file to remove explicit protocol listings and rely on the plugin for navigation.
4. Test the changes to ensure they work as expected.

I'll now proceed with implementing these changes.

uthentic-ai commented:
I have updated the PR with the following changes:

1. Updated the protocol_nav_plugin.py to ensure the correct URI structure for protocols (e.g., /action-menu/1.0/ instead of /protocols/action-menu/1.0/).
2. Added debug output to the plugin to inspect the nav before and after the plugin runs.
3. Updated the mkdocs.yml file to remove explicit protocol listings and rely on the plugin for navigation.
4. Fixed an issue in the _log_nav method of the plugin to correctly handle both Navigation and Section objects.

The site now builds successfully with the new structure and custom plugin. The protocol pages are now served with the appropriate URI structure, and new protocols can be added without explicitly including them in the mkdocs.yml file.

There are still some warnings about missing images and broken links, which were previously documented in the CONVERSION_SUMMARY.md file. These issues don't prevent the site from building but should be addressed in future updates to improve documentation quality.

Please review these changes and let me know if any further modifications are needed.
